genie-agent/src/main/java/com/netflix/genie/agent/execution/services/AgentFileManifestService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import java.io.File;
+
+/**
+ * This service regularly produces a manifest of the executing job folder and pushes it to the server.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public interface AgentFileManifestService {
+    /**
+     * Start the service.
+     *
+     * @param claimedJobId the claimed job id
+     * @param jobDirectory the job directory
+     */
+    void start(String claimedJobId, File jobDirectory);
```

Comment:
Would prefer use of `java.nio.Path`

Reply:
Switched both to `Path` and created a JIRA to move other things from `File` to `Path` (such as the job directory in the `JobSpecification` POJO, which is where this comes from.